### DISCUSSION TITLE
[QUESTION] kubernetes.io/csi: failed to open volume data file

### DISCUSSION THREAD

**Discussion by tomsunyu** [longhorn/longhorn#5767] [Q&A]
Question: I found logs such as:

```
Apr 15 14:44:27 k8sn12 kubelet: E0415 14:44:27.830417    9004 reconciler.go:193] "operationExecutor.UnmountVolume failed (controllerAttachDetachEnabled true) for volume \"data-mysql\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5\") pod \"6f4d9977-45d0-41f6-8a35-f948501f721a\" (UID: \"6f4d9977-45d0-41f6-8a35-f948501f721a\") : UnmountVolume.NewUnmounter failed for volume \"data-mysql\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5\") pod \"6f4d9977-45d0-41f6-8a35-f948501f721a\" (UID: \"6f4d9977-45d0-41f6-8a35-f948501f721a\") : kubernetes.io/csi: unmounter failed to load volume data file [/var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/mount]: kubernetes.io/csi: failed to open volume data file [/var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/vol_data.json]: open /var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/vol_data.json: no such file or directory" err="UnmountVolume.NewUnmounter failed for volume \"data-mysql\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5\") pod \"6f4d9977-45d0-41f6-8a35-f948501f721a\" (UID: \"6f4d9977-45d0-41f6-8a35-f948501f721a\") : kubernetes.io/csi: unmounter failed to load volume data file [/var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/mount]: kubernetes.io/csi: failed to open volume data file [/var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/vol_data.json]: open /var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/vol_data.json: no such file or directory"
```
My pod is running.
```
[root@k8sm1 ~]# kubectl get po -n people
NAME                                       READY   STATUS    RESTARTS   AGE
mysql-0                                    1/1     Running   0          7d1h
```

My PVC like this:
```
[root@k8sm1 ~]# kubectl get pvc -n people
NAME                                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-mysql-mysql-0                  Bound    pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5   1Gi        RWO            longhorn       23d
```

Through the logs, I attempted to find the path to the mount， I discovered these issues
```
[root@k8sn12 ~]# ls /var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5
ls: cannot access /var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/mount: Input/output error
```

I don't know why this error is reported in the log. Can you help me ?

**Comment by c3y1huang** [comment]
Some possible related issues:
- https://github.com/longhorn/longhorn/issues/2368
- https://github.com/longhorn/longhorn/issues/4782

@tomsunyu , are you able to provide some context about your environment? Also maybe share reproducing steps?

**Comment by tomsunyu** [comment]
> Some possible related issues:
> 
>     * [[BUG] Observed massive `CSI UnmountVolume failed` in the kubelet log #2368](https://github.com/longhorn/longhorn/issues/2368)
> 
>     * [[BUG] Restarting kubelet and upgrading longhorn pod get stuck in terminating #4782](https://github.com/longhorn/longhorn/issues/4782)
> 
> 
> @tomsunyu , are you able to provide some context about your environment? Also maybe share reproducing steps?

### Environment:
* Longhorn version: v1.3.1
* Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:  Kubespray deploy
  * Number of management node in the cluster: 3
  * Number of worker node in the cluster: 16
* Node config
   * OS type and version: CentOS7.9
   * CPU per node: 16
   * Memory per node: 32Gi
* Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): KVM

**Comment by tomsunyu** [comment]
I manually unmounted this volume: /var/lib/kubelet/pods/6f4d9977-45d0-41f6-8a35-f948501f721a/volumes/kubernetes.io~csi/pvc-8edc8e20-b814-4ff4-b7ec-3a6ceb8dbdf5/mount, and no longer generating error logs.
But is this the correct solution?

**Reply by c3y1huang** [reply]
What is your K8s version?

It may be worth checking this upstream. [This issue](https://github.com/kubernetes/kubernetes/issues/116847) seems currently active and related.

**Reply by tomsunyu** [reply]
> What is your K8s version?
> 
> It may be worth checking this upstream. [This issue](https://github.com/kubernetes/kubernetes/issues/116847) seems currently active and related.

v1.21.0
